builtin/credential/jwt/path_cel_role.go

```diff
@@ -0,0 +1,485 @@
+	roleName := data.Get("name").(string)
+	if roleName == "" {
+		return logical.ErrorResponse("missing CEL role name"), nil
+	}
+
+	role, err := b.getCelRole(ctx, req.Storage, roleName)
+	if err != nil || role == nil {
+		return nil, err
+	}
+
+	resp := &logical.Response{
+		Data: role.ToResponseData(),
+	}
+	return resp, nil
+}
+
+func (b *jwtAuthBackend) pathCelRolePatch(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	txRollback, err := logical.StartTxStorage(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer txRollback()
+
+	roleName := data.Get("name").(string)
+
+	oldEntry, err := b.getCelRole(ctx, req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if oldEntry == nil {
+		return logical.ErrorResponse("Unable to fetch cel role entry to patch"), nil
+	}
+
+	entry := &celRoleEntry{
+		Name:          roleName,
+		CelProgram:    data.GetWithExplicitDefault("cel_program", oldEntry.CelProgram).(celhelper.CelProgram),
+		FailurePolicy: data.GetWithExplicitDefault("failure_policy", oldEntry.FailurePolicy).(string),
+		Message:       data.GetWithExplicitDefault("message", oldEntry.Message).(string),
+	}
+
+	resp, err := validateCelRoleCreation(b, entry, ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if resp.IsError() {
+		return resp, nil
+	}
+
+	// Store it
+	jsonEntry, err := logical.StorageEntryJSON("cel/role/"+roleName, entry)
+	if err != nil {
+		return nil, err
+	}
+	if err := req.Storage.Put(ctx, jsonEntry); err != nil {
+		return nil, err
+	}
+
+	if err := logical.EndTxStorage(ctx, req); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (b *jwtAuthBackend) pathCelRoleDelete(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	err := req.Storage.Delete(ctx, "cel/role/"+data.Get("name").(string))
+	return nil, err
+}
+
+func (b *jwtAuthBackend) getCelRole(ctx context.Context, s logical.Storage, roleName string) (*celRoleEntry, error) {
+	entry, err := s.Get(ctx, "cel/role/"+roleName)
+	if err != nil || entry == nil {
+		return nil, err
+	}
+
+	var result celRoleEntry
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	result.Name = roleName
+
+	return &result, nil
+}
+
+func validateCelRoleCreation(b *jwtAuthBackend, entry *celRoleEntry, ctx context.Context, s logical.Storage) (*logical.Response, error) {
+	resp := &logical.Response{}
+
+	_, err := b.validateCelProgram(entry.CelProgram)
+	if err != nil {
+		return nil, fmt.Errorf("%w", err)
+	}
+
+	resp.Data = entry.ToResponseData()
+	return resp, nil
+}
+
+func (b *jwtAuthBackend) validateCelProgram(program celhelper.CelProgram) (bool, error) {
+	// adding a minimal jwtClaims collection here, for validating usages in CEL expression
+	_, err := b.celEvalProgram(program, map[string]any{"sub": "email@example.com", "aud": "audience", "iss": "issuer"})
```

Comment:
:thinking: Hmm, I do wonder how scalable this will be, but I suppose enforcing optionality for some of these might be hard.